### ISSUE TITLE
proposal:  keep "db" and "passwd" around as long term "legacy" keywords, remove warnings

### ISSUE THREAD

**Issue by zzzeek** [PyMySQL/PyMySQL#939]
pymysql was originally began as a front-to-back compatible pure Python port of python-mysql.   python-mysql was then reworked as mysqlclient underneath the same pymysql port.

one of the main draws of mysqlclient and pymysql is that they maintain the original python-mysql API, as unpolished as it was.    The "db" and "passwd" parameters have been the same way for at least fifteen years.

I'm all for adding nicer parameters "database" and "password", but the change in pymysql to have "db" and "passwd" actually emit a warning message is going to produce a significant amount of work for people, as "warnings" are more often than not failure conditions for test suites not to mention most users report them as errors in any case.

in this case the warnings were just added in 511b6a2a two days ago and already released, not even fast enough for me to notice our DBAPI master builds failing.

I would propose simply removing these warnings and leaving "passwd" and "db" around long term as they are not adding complexity to pymysql.    in SQLAlchemy we have a lot of keywords like this where we recognize that some name changes are not worth removing the old names, which don't actually get in the way of anything.

thanks for reading!

**Comment by methane** [comment]
First of all, `database` and `password` are recommended by the DB-API 2.0. They are used in `my.cnf` too. So `db` and `passwd` is just a garbage, useless name. Such names are just causing confusion.

> I would propose simply removing these warnings and leaving "passwd" and "db" around long term as they are not adding complexity to pymysql. 

They are adding maintenance costs, including mental costs. And they have been leaved "long term" (7 years!) already. That's why I added the warning.

Generally speaking, adding warning is not a breaking change. It is just a warning for the future change.

But for SQLAlchemy, I decided to remove the warning. The parameters are still deprecated, but no warning at least one more year.

**Comment by zzzeek** [comment]
> First of all, `database` and `password` are recommended by the DB-API 2.0. They are used in `my.cnf` too. So `db` and `passwd` is just a garbage, useless name. Such names are just causing confusion.

I agree they are "garbage", they always have been.    If they are off to the side and not documented, probably not very confusing.    in SQLAlchemy I will often un-document an old API that nobody should be using but if it does not complicate any internals I'll leave it around for a long time.   It looks like "database" and "password" have been accepted for years but i wasn't aware of that.   mysqlclient still documents "db" and "passwd".

> 
> > I would propose simply removing these warnings and leaving "passwd" and "db" around long term as they are not adding complexity to pymysql.
> 
> They are adding maintenance costs, including mental costs. And they have been leaved "long term" (7 years!) already. That's why I added the warning.
> 
> Generally speaking, adding warning is not a breaking change. It is just a warning for the future change.

it broke all our builds since our test suites raise for any unexpected warnings; these are bugs for us because any warnings that come out are reported by users as bugs, and these days a lot of users don't know the difference between errors and warnings and think their program is not working correctly.   this is the current state of things.   

We have a "DBAPI master" build that runs SQLAlchemy's test suite against the git master of most of the major DBAPIs, but in this case, the change was committed and released as 1.0 within just a few days.   so we did not catch it in time.    The "drop python 2 support", see that was in there for a long time so we were ready for that (actually we werent but I knew about it for a long time so that one's on me).    

At the same time as we got the warnings, aiomysql had breakage as well as they are using too many of pymysql's internals that also broke our builds (they are also ignoring my PRs, which is bad), which is on them, but that made this seem like a larger scale failure than it was.     

> 
> But for SQLAlchemy, I decided to remove the warning. The parameters are still deprecated, but no warning at least one more year.

Thanks for that!   Surprisingly, I didnt get bug reports for this today, usually I do.   We have already committed fixes for the new names so either way. 

I would like to know if there are plans to change mysqlclient as well.  It looks like "database" is accepted but the [docs](https://mysqlclient.readthedocs.io/user_guide.html#functions-and-attributes) still use "db" and "passwd".  I can change those to use "database" and "password" from SQLAlchemy if myqslclient is going to change them.

thanks!
